experimental/framework/fluid-static/src/FluidStatic.ts

```diff
@@ -4,64 +4,129 @@
  */
 
 import { getContainer, IGetContainerService } from "@fluid-experimental/get-container";
-import { getObjectWithIdFromContainer } from "@fluidframework/aqueduct";
+import { DataObject, getObjectWithIdFromContainer } from "@fluidframework/aqueduct";
 import { IContainer } from "@fluidframework/container-definitions";
-import { IFluidDataStoreFactory, NamedFluidDataStoreRegistryEntry } from "@fluidframework/runtime-definitions";
-import { DOProviderContainerRuntimeFactory } from "./containerCode";
+import { NamedFluidDataStoreRegistryEntry } from "@fluidframework/runtime-definitions";
+import { DOProviderContainerRuntimeFactory, IFluidStaticDataObjectClass } from "./containerCode";
 
-export interface IFluidStaticDataObjectClass {
-    readonly factory: IFluidDataStoreFactory;
+export interface FluidContainerConfig {
+    id: string,
+    dataObjects: IFluidStaticDataObjectClass[],
+}
+
+export interface FluidCreateContainerConfig extends FluidContainerConfig {
+    /**
+     * initial DataObjects will be created when the Container is first created.
+     */
+    initialDataObjects?: [string, IFluidStaticDataObjectClass][]
 }
 
 export class FluidContainer {
-    constructor(private readonly container: IContainer, public readonly createNew: boolean) { }
+    private readonly types: Set<string>;
+    constructor(
+        private readonly container: IContainer,
+        namedRegistryEntries: NamedFluidDataStoreRegistryEntry[],
+        public readonly createNew: boolean) {
+        this.types = new Set();
+        namedRegistryEntries.forEach((value: NamedFluidDataStoreRegistryEntry) => {
+            const type = value[0];
+            if (this.types.has(type)) {
+                throw new Error(`Multiple DataObjects share the same type identifier ${value}`);
+            }
+            this.types.add(type);
+        });
+    }
 
-    public async createDataObject<T = any>(dataObjectClass: IFluidStaticDataObjectClass, id: string) {
+    public async createDataObject<T extends DataObject>(
+        dataObjectClass: IFluidStaticDataObjectClass, id: string): Promise<T> {
         const type = dataObjectClass.factory.type;
+        // This is a runtime check to ensure the developer doesn't try to create something they have not defined.
+        if (!this.types.has(type)) {
+            throw new Error("Trying to create a DataObject that was not defined in the Fluid.init");
+        }
+
         await this.container.request({ url: `/create/${type}/${id}` });
         const dataObject = await this.getDataObject<T>(id);
         return dataObject;
     }
 
-    public async getDataObject<T = any>(id: string) {
+    public async getDataObject<T extends DataObject>(id: string): Promise<T> {
```

Comment:
Not all data objects are `DataObject`s, right?  Is there an interface we can use here instead of `DataObject`?

The underlying implementation also doesn't actually make any guarantees that a DataObject is returned, e.g. if I wanted to get wild maybe I call `getDataObject("_scheduler")`

Reply:
I think we should make a strong assertion that the consumption model only works with classes that extend the `DataObject` class and we do not support developers ability to use the framework to build other thing.

We've seen in the 1P scenario that the ability to use Fluid and DataStore factories to return anything has lead to confusion and has never successfully worked. 

I believe if this is a scenario the developer wants we they have moved to ultra advanced scenarios and are not longer in the world of the consumption model.

I do agree that there is no guarantee and I would love to think more about how to get there. I do think there is a loose guarantee around the fact that we only let developers create DataObjects.